Document dependency verification update process (#18863)
gradle/gradle: subprojects/docs/src/docs/userguide/dep-man/01-core-dependency-management/dependency_verification.adoc

Only plugins which use the infrastructure provided by Gradle for performing requests will see their requests verified.
====

[[sec:verification-update]]
=== Using generation for incremental updates

The verification file generated by Gradle has a strict ordering for all its content.
It also uses the information from the existing state to limit changes to the strict minimum.

This means that generation is actually a convenient tool for _updating_ a verification file:
* Checksum entries generated by Gradle will have a clear `origin` that starts with "Generated by Gradle", which is a good indicator that an entry needs to be reviewed,

> * Entries added by hand will immediately be accounted for, and appear at their right location after writing the file,

Reviewer: ```suggestion
* Entries added by hand will immediately be accounted for, and appear at the right location after writing the file,
```

> * The header comments of the file will be preserved, that is comments before the root XML node.

Reviewer: ```suggestion
* The header comments of the file will be preserved, i.e. comments before the root XML node.
```

This allows to have license or instructions on which tasks and which parameters to use for generating that file.

With the above benefits, it is really easy to account for new dependencies or dependency versions by simply generating the file again and reviewing the changes.

> This behaviour should make the feature much easier to use in practice.

Reviewer 2: I agree.  This: "This behaviour should make the feature much easier to use in practice." should be removed from this part of the documentaion.
